### PR TITLE
Bug #630 population folder

### PR DESCRIPTION
**Pull request by MichaelStritt**
### Linked issue

Check out #630

### CODE REVIEW

**Review by jan-petr — APPROVED** [review]
Correct reorganization of the code. But it still creates the Population file in the derivatives folder. Was that the problem - created on the correct location but too early? @BeatrizPadrela 

Anyway - I removed the initFileSystem function from ExploreASL_Initialize and moved it only to structural/asl/population module. So this should be now fixed, but Henk should check as well...

**Review by HenkMutsaerts — CHANGES_REQUESTED** [review]
Yes, you're right, the idea is that the population and its subdirs are only created when needed. So this fix is nice, but even nicer to remove this folder creation out of general files and manage this within the individual wrappers that need these folders to exist. 

The issue was not `xASL_init_FileSystem` but `xASL_init_DefinePaths` I believe, that creates everything upon initialization. Easiest fix for now is to move the contents of `xASL_init_DefinePaths` from line 81 onward (loop of `xASL_adm_CreateDir`) to the start of `ExploreASL_ProcessMaster`.

When data is loaded, it should define the paths, when data is processed (i.e., running `ExploreASL_ProcessMaster`) it should create the folders. Is this the case now, that `ExploreASL_ProcessMaster` is only run when processing is set to true, and this creates the folders?

File: Modules/xASL_module_Population.m [hunk 1]
```diff
@@ -35,6 +35,12 @@
 %% ------------------------------------------------------------------------------------------------------------
 %% Admin
 
+% Create population directory
+if ~xASL_exist(x.D.PopDir)
```
[the remark this answers is not in the document]
Comment (jan-petr) [reply Modules/xASL_module_Population.m]: Yes. Fixed.

**Comment by jan-petr** [comment]
> Yes, you're right, the idea is that the population and its subdirs are only created when needed. So this fix is nice, but even nicer to remove this folder creation out of general files and manage this within the individual wrappers that need these folders to exist.
> 
> The issue was not `xASL_init_FileSystem` but `xASL_init_DefinePaths` I believe, that creates everything upon initialization. Easiest fix for now is to move the contents of `xASL_init_DefinePaths` from line 81 onward (loop of `xASL_adm_CreateDir`) to the start of `ExploreASL_ProcessMaster`.
> 
> When data is loaded, it should define the paths, when data is processed (i.e., running `ExploreASL_ProcessMaster`) it should create the folders. Is this the case now, that `ExploreASL_ProcessMaster` is only run when processing is set to true, and this creates the folders?

I agree in principle. `xASL_init_DefinePaths` should not create anything - but this is already fixed by Michael. All directory creation is done in `xASL_init_FileSystem` - this is then called from all modules (ASL, structural, etc.). So it is now indeed only run when the processind modules are run. It might be possible to move all this from these modules to `ExploreASL_ProcessMaster` if needed - but not sure if this wouldn't have any complications.

Anyway - the things already seem to be done according to what we want... Can we merge?

**Review by HenkMutsaerts — APPROVED** [review]
Yes this looks good. If this is tested and works, I'm good and can be merged. I would keep `xASL_init_FileSystem` in the modules for now to keep modularity.

**Pull request merged**